### PR TITLE
DM-13768 post-review fixes for headers, making https:// work

### PR DESCRIPTION
**Pull request by stargaser**
**This is a preliminary pull request, not yet ready for final review**

Handle some issues with headers that came up when testing the `display_firefly` part of DM-13768.

* bump bug fix version to 1.2.2
* populate headers in FireflyClient

### CODE REVIEW

**Comment by stargaser** [comment]
@robyww @tgoldina -- I've tried an initialization of headers in this PR. It works fine with the lone exception of http protocol, `channel=None` where I have found I can init a `FireflyClient` and then ask for `get_firefly_url`, and create a never-ending loop where `channel` is never populated.

Maybe it is better for the client to handle the `channel=None` case itself without waiting for `channel` to be set by the server? It bothers me that it is not always possible to ask the `FireflyClient` immediately for the url for the browser.

**Comment by tgoldina** [comment]
> Maybe it is better for the client to handle the channel=None case itself without waiting for channel to be set by the server?

Yes, I thought that's what we agreed on yesterday. If channel is None, we'll just come with some "unique" string for it. For example, a timestamp + some random number can make it unique enough.

Perhaps you have missed the last message from Trey, 3:33p yesterday.

File: firefly_client/firefly_client.py [hunk 1]
```diff
@@ -119,6 +119,7 @@ def __init__(self, host=_my_localhost, channel=None, basedir='firefly', html_fil
         self.url_bw = protocol + '://' + self.this_host + '/%s%s?__wsch=' % (self._basedir, self.html_file)
         self.listeners = {}
         self.channel = channel
+        self.headers = {'FF-channel':channel} if channel else {}
```
Comment (tgoldina) [inline firefly_client/firefly_client.py]: To make sure channel is always set, can we use the default channel=str(uuid.uuid1()) rather than channel=None?

**Review by tgoldina — APPROVED** [review]
Looks good to me.

**Pull request merged**